Maw annotate text

brownbrain opened the pull request:
Please review the code changes. I added the functionality and tests for annotating text.

brownbrain committed:
Fixed building issues with PDFBox

Updated the font archive urls

brownbrain committed:
Annotate Text Functionality

Added AddText class for annotating text, and TestAddText class to test the annotation of text.

A-Cain commented on tools/src/test/java/org/apache/pdfbox/tools/TestAddText.java:
```diff
@@ -0,0 +1,57 @@
+        String expected = "../tools/src/main/java/org/apache/pdfbox/tools/AddImageResources/examplePDFWithText.pdf";
+
+
+        try
```
Is there any reason this is a try catch instead of having the function throw an exception?

A-Cain commented on tools/src/test/java/org/apache/pdfbox/tools/TestAddText.java:
```diff
@@ -0,0 +1,57 @@
+    {
+        AddText add = new AddText();
+        
+        String expected = "../tools/src/main/java/org/apache/pdfbox/tools/AddImageResources/examplePDFWithText.pdf";
```
I would like it if this test did not use the same folder as AddImage for its tests.

A-Cain commented on tools/src/test/java/org/apache/pdfbox/tools/TestAddText.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    @Test
+    public void testWriteText()
```
When I run this on my own machine, the file "examplePDFWithText" does not contain any additional text

A-Cain reviewed (changes requested):
The file the test cases modify does not contain any additional text after I run it.
Additionally, I would like it if it had its own test resources directory that is not shared with AddImage

brownbrain committed:
Fixed Adding Text

Fixed the AddText class. Annotations now properly appear in pdf documents after method is called.

Also, copied test pdfs from the AddImageResources directory to a new AddTextResources directory to prevent sharing.

A-Cain commented on tools/src/test/java/org/apache/pdfbox/tools/TestAddText.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    @Test
+    public void testWriteText()
```
This test fails when I run it on my machine. 

junit.framework.AssertionFailedError: 
Expected :1
Actual   :0

Not sure what you're trying to assert at the end of this test. The actual you entered is 0 which will never pass because 0 never equals 1

A-Cain reviewed (approved):
Looks all good to me. Tests pass on my machine

brownbrain committed:
Delete examplePDFWithTextAndImage.pdf
